src/test/boa/test/datagen/TestBuildSnapshot.java

```diff
@@ -213,6 +213,18 @@ static CodeRepository buildCodeRepository(String repoName) throws Exception {
 			assertArrayEquals(expectedFileNames, fileNames);
 		}
 		
+		// test changed files for each commit
+		for (Revision rev : cr.getRevisionsList()) {
+			String[] expectedFileNames = conn.getDiffFiles(rev.getId()).toArray(new String[0]);
+			String[] actualFileNames = new String[rev.getFilesCount()];
+			for (int i = 0; i < actualFileNames.length; i++)
+				actualFileNames[i] = rev.getFiles(i).getName();
+			Arrays.sort(expectedFileNames);
+			Arrays.sort(actualFileNames);
+			System.out.println(rev.getId());
+			assertArrayEquals(expectedFileNames, actualFileNames);
+		}
+		

```

Comment:
The previous tests only consider the snapshot for each commit. So, we need to add the test case for checking changed files saved under each commit.